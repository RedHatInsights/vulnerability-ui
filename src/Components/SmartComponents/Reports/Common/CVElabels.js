import React from 'react';
import propTypes from 'prop-types';
import { View } from '@react-pdf/renderer';
import { InsightsLabel } from '@redhat-cloud-services/frontend-components-pdf-generator';

import styles from './styles';
import messages from '../../../../Messages';

const CVElabels = ({ intl, hasRule, hasExploit, isSmall = false }) => (
    <View style={styles.labelsWrapper}>
        {hasExploit &&
            <View style={{ ...styles.label, width: isSmall ? 62 : 70 }}>
                <InsightsLabel
                    icon={false}
                    width={isSmall ? 62 : 75}
                    variant={4}
                    label={intl.formatMessage(messages.knownExploit)}
                    textStyle={isSmall ? { ...styles.smallText, left: -56 } : undefined}
                    labelStyle={{
                        border: 'solid',
                        borderWidth: 1,
                        borderColor: '#c9190b',
                        height: isSmall ? 12 : 20
                    }}
                />
            </View>
        }
        {hasRule &&
            <View style={styles.label}>
                <InsightsLabel
                    icon={false}
                    width={isSmall ? 59 : 63}
                    label={intl.formatMessage(messages.securityRuleLabel)}
                    textStyle={isSmall ? styles.smallText : undefined}
                    labelStyle={{
                        border: 'solid',
                        borderWidth: 1,
                        borderColor: '#bee1f4',
                        height: isSmall ? 12 : 20
                    }}
                />
            </View>
        }
    </View>
);

CVElabels.propTypes = {
    hasRule: propTypes.bool,
    hasExploit: propTypes.bool,
    intl: propTypes.any,
    isSmall: propTypes.bool
};

export default CVElabels;
