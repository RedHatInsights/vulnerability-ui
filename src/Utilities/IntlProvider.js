import { createIntl, createIntlCache } from 'react-intl';

const cache = createIntlCache();
const locale = navigator.language.slice(0, 2);

export const intl = createIntl({
    locale
}, cache);

export const intlSettings = { locale };
