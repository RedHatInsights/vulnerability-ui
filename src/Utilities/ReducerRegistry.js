import promiseMiddleware from 'redux-promise-middleware';
import { getRegistry } from '@red-hat-insights/insights-frontend-components';

const registry = getRegistry({}, [promiseMiddleware()]);

export default {
    getStore: () => registry.getStore(),
    register: (newReducers) => {
        registry.register(newReducers);
    }
};
