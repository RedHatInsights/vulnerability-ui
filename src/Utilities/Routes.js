import React from 'react';
import some from 'lodash/some';
import PropTypes from 'prop-types';
import { Redirect, Route, Switch, useLocation } from 'react-router-dom';
import asyncComponent from './asyncComponent';
import { PATHS } from '../Helpers/constants';

const SystemsPage = asyncComponent(() =>
    import(
        /* webpackChunkName: "Systems" */ '../Components/SmartComponents/SystemsPage/SystemsPage'
    )
);

const LandingPage = asyncComponent(() =>
    import(
        /* webpackChunkName: "LandingPage" */ '../Components/SmartComponents/LandingPage/LandingPage'
    )
);

const Upgrade = asyncComponent(() =>
    import(
        /* webpackChunkName: "Upgrade" */ '../Components/PresentationalComponents/StaticPages/UpgradePage'
    )
);

const CVEDetailsPage = asyncComponent(() =>
    import(
        /* webpackChunkName: "CVEDetailsPage" */ '../Components/SmartComponents/CVEDetailsPage/CVEDetailsPage'
    )
);

const SystemDetailsPage = asyncComponent(() =>
    import(
        /* webpackChunkName: "SystemDetailsPage" */ '../Components/SmartComponents/SystemDetailsPage/SystemDetailsPage'
    )
);
const Reports = asyncComponent(() =>
    import(/* webpackChunkName: "Reports" */ '../Components/SmartComponents/Reports/ReportsPage')
);

const InsightsRoute = ({ component: Component, rootClass = 'vulnerability', ...rest }) => {
    const root = document.getElementById('root');
    root.removeAttribute('class');

    root.classList.add(`page__${rootClass}`, 'pf-l-page__main');
    root.classList.add(`page__${rootClass}`, 'pf-c-page__main');
    root.setAttribute('role', 'main');

    return <Route {...rest} component={Component} />;
};

InsightsRoute.propTypes = {
    component: PropTypes.func,
    rootClass: PropTypes.string
};

export const Routes = () => {
    let location = useLocation();
    let path = location.pathname;
    return (
        <Switch>
            <Redirect from={PATHS.cvepagePath.to} to={PATHS.systemDetailsPage.to} />
            <InsightsRoute exact path={PATHS.cveDetailsPage.to} component={CVEDetailsPage} />
            <InsightsRoute exact path={PATHS.systemDetailsPage.to}  component={SystemDetailsPage} />
            <InsightsRoute exact path={PATHS.cvesPage.to} component={LandingPage} />
            <InsightsRoute exact path={PATHS.systemsPage.to} component={SystemsPage} />
            <InsightsRoute exact path={PATHS.upgrade.to} component={Upgrade} />
            <InsightsRoute exact path={PATHS.reports.to} component={Reports} />
            <Route
                render={() => (some(PATHS, p => p.to === path) ? null : <Redirect to={PATHS.cvesPage.to} />)}
            />
        </Switch>
    );
};

