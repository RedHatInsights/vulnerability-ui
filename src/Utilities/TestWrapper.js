import propTypes from 'prop-types';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

const TestWrapper = ({ children, store }) => {
    return (
        <IntlProvider locale="en">
            { store ?
                <Provider store={store}>
                    <Router>
                        {children}
                    </Router>
                </Provider> : children
            }
        </IntlProvider>
    );
};

TestWrapper.propTypes = {
    children: propTypes.node,
    store: propTypes.object
};

export default TestWrapper;
