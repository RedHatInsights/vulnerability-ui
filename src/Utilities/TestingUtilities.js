import React from 'react';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { IntlProvider } from 'react-intl';

export const ComponentWithContext = ({
    renderOptions = {},
    children
}) => {
    const mockStore = configureStore();

    return (
        <IntlProvider locale="en">
            <Provider store={renderOptions?.store || mockStore()}>
                <MemoryRouter initialEntries={renderOptions?.initialEntries || ['/']}>
                    {
                        renderOptions?.componentPath ? (
                            <Routes>
                                <Route>
                                    {children}
                                </Route>
                            </Routes>
                        ) : children
                    }
                </MemoryRouter>
            </Provider>
        </IntlProvider>
    );
};

ComponentWithContext.propTypes = {
    children: PropTypes.element,
    renderOptions: PropTypes.object
};
