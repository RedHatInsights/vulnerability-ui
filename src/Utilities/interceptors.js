import {
    authInterceptor,
    interceptor401,
    responseDataInterceptor
} from '@redhat-cloud-services/frontend-components-utilities/interceptors';
import axios from 'axios';

const instance = axios.create();

export function errorInterceptor(error) {
    if (!axios.isCancel(error)) {
        const errObject = { ...error };

        errObject?.response?.status === 503 && window.setReadOnlyBannerVisible?.(true);

        if (errObject?.response?.data?.errors) {
            let data = errObject.response.data.errors[0];

            if (typeof data.detail === 'object') {
                data.detail = data.detail.msg;
            }

            throw data;
        }

        throw error;
    }
}

const readOnlyBannerInterceptor = response => {
    // if has updated property, then it was a successful write request
    response.hasOwnProperty('updated') && window.setReadOnlyBannerVisible?.(false);
    return response;
};

instance.interceptors.request.use(authInterceptor);
instance.interceptors.response.use(responseDataInterceptor);
instance.interceptors.response.use(null, interceptor401);
instance.interceptors.response.use(readOnlyBannerInterceptor, errorInterceptor);

export default instance;
