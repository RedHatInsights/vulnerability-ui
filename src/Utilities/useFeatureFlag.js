import { useFlag, useFlagsStatus } from '@unleash/proxy-client-react';

export default (flag) => {
    const { flagsReady } = useFlagsStatus();
    const isFlagEnabled = useFlag(flag);
    return flagsReady ? isFlagEnabled : false;
};
