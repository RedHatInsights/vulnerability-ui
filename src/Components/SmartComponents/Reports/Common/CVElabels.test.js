import CVElabels from './CVElabels';
import configureStore from 'redux-mock-store';
import { intl } from '../../../../Utilities/IntlProvider';
import { render } from '@testing-library/react';

const mockStore = configureStore([store => next => action => {}]);
let store = mockStore({ });

describe('CVElabels component', () => {
    it.skip('Should match snapshot with no labels.', () => {
        const { asFragment } = render(
            <CVElabels intl={intl} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it.skip('Should match snapshot with exploit and security rule label.', () => {
        const { asFragment } = render(
            <CVElabels intl={intl} hasRule hasExploit />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
