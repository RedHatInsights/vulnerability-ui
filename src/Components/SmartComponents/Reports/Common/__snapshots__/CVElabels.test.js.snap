// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVElabels component Should match snapshot with exploit and security rule label. 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <CVElabels
        hasExploit={true}
        hasRule={true}
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "defaultRichTextElements": undefined,
            "formatDate": [Function],
            "formatDateTimeRange": [Function],
            "formatDateToParts": [Function],
            "formatDisplayName": [Function],
            "formatList": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatNumberToParts": [Function],
            "formatPlural": [Function],
            "formatRelativeTime": [Function],
            "formatTime": [Function],
            "formatTimeToParts": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getDisplayNames": [Function],
              "getListFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralRules": [Function],
              "getRelativeTimeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "onError": [Function],
            "textComponent": Symbol(react.fragment),
            "timeZone": undefined,
          }
        }
      >
        <VIEW
          style={
            Object {
              "flexDirection": "row",
              "marginBottom": "2px",
              "marginTop": "2px",
            }
          }
        >
          <VIEW
            style={
              Object {
                "marginRight": "8px",
                "width": 70,
              }
            }
          >
            <InsightsLabel
              icon={false}
              label="Known exploit"
              variant={4}
              width={75}
            >
              <VIEW
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
                width={75}
              >
                <CANVAS
                  style={
                    Object {
                      "backgroundColor": "#faeae8",
                      "borderRadius": 30,
                      "height": 20,
                      "width": 75,
                    }
                  }
                />
                <TEXT
                  style={
                    Object {
                      "color": "#7d1007",
                      "left": -65,
                      "top": 4,
                    }
                  }
                >
                  Known exploit
                </TEXT>
              </VIEW>
            </InsightsLabel>
          </VIEW>
          <VIEW
            style={
              Object {
                "marginRight": "8px",
                "width": 70,
              }
            }
          >
            <InsightsLabel
              icon={false}
              label="Security rule "
              variant={1}
              width={63}
            >
              <VIEW
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
                width={63}
              >
                <CANVAS
                  style={
                    Object {
                      "backgroundColor": "#e7f1fa",
                      "borderRadius": 30,
                      "height": 20,
                      "width": 63,
                    }
                  }
                />
                <TEXT
                  style={
                    Object {
                      "color": "#002952",
                      "left": -56,
                      "top": 4,
                    }
                  }
                >
                  Security rule 
                </TEXT>
              </VIEW>
            </InsightsLabel>
          </VIEW>
        </VIEW>
      </CVElabels>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`CVElabels component Should match snapshot with no labels. 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <CVElabels
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "defaultRichTextElements": undefined,
            "formatDate": [Function],
            "formatDateTimeRange": [Function],
            "formatDateToParts": [Function],
            "formatDisplayName": [Function],
            "formatList": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatNumberToParts": [Function],
            "formatPlural": [Function],
            "formatRelativeTime": [Function],
            "formatTime": [Function],
            "formatTimeToParts": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getDisplayNames": [Function],
              "getListFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralRules": [Function],
              "getRelativeTimeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "onError": [Function],
            "textComponent": Symbol(react.fragment),
            "timeZone": undefined,
          }
        }
      >
        <VIEW
          style={
            Object {
              "flexDirection": "row",
              "marginBottom": "2px",
              "marginTop": "2px",
            }
          }
        />
      </CVElabels>
    </Router>
  </BrowserRouter>
</Provider>
`;
